adapters/adapters-base/src/main/java/org/apache/cassandra/sidecar/adapters/base/CassandraStorageOperations.java

```diff
@@ -132,10 +133,19 @@ protected void takeSnapshotInternal(@NotNull String tag,
         }
         catch (IOException e)
         {
+            // post-5.0, IOExceptions are thrown when previously something else like
+            // an IllegalArgumentException was thrown. First, try to unwrap the IOException and throw
+            // the original cause, which we will process correctly in CreateSnapshotHandler
+            // if the exception was an IllegalArgumentException
+            IllegalArgumentException iex = ThrowableUtils.getCause(e, IllegalArgumentException.class);
+            if (iex != null) {
```

Comment:
the formatting seems off in some places
```suggestion
            if (iex != null)
            {
```